Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Forward.h>
+#include <AK/HashTable.h>
+#include <AK/SourceGenerator.h>
+#include <AK/String.h>
+#include <AK/Try.h>
+#include <AK/Utf8View.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <LibGUI/GML/Parser.h>
+#include <LibGUI/UIDimensions.h>
+#include <LibMain/Main.h>
+
+enum class UseObjectConstructor : bool {
+    No,
+    Yes,
+};
+
+static ErrorOr<void> generate_loader_for_object(GUI::GML::Object const& gml_object, SourceGenerator generator, String object_name, size_t indentation, UseObjectConstructor use_object_constructor);
```

Comment:
Is this declaration needed? It seems like the next usage of `generate_loader_for_object` will be the definition itself either way.

Reply:
Huh, I thought it wasn't.